pkg/plugins/runtime/k8s/webhooks/injector/injector.go

```diff
@@ -213,6 +226,62 @@ func (i *KumaInjector) meshFor(pod *kube_core.Pod, ns *kube_core.Namespace) (*co
 	return meshResource, nil
 }
 
+// loadCustomPatches loads the ContainerPatch CRDs associated with the given pod, divides out
+// the sidecar and init patches, and concatinates each type into its own list for return.
+func (i *KumaInjector) loadCustomPatches(pod *kube_core.Pod, ns *kube_core.Namespace) (sidecarPatches []mesh_k8s.JsonPatchBlock, initPatches []mesh_k8s.JsonPatchBlock, err error) {
+	patchNames := i.cfg.ContainerPatches
+	if val, exist := metadata.Annotations(pod.Annotations).GetString(metadata.KumaContainerPatches); exist {
+		patchNames = val
+	}
+
+	if patchNames == "" {
+		// Avoid Split returning slice containing empty string
+		return []mesh_k8s.JsonPatchBlock{}, []mesh_k8s.JsonPatchBlock{}, nil
+	}
+
+	for _, patchName := range strings.Split(patchNames, ",") {
+		containerPatch := &mesh_k8s.ContainerPatch{}
+		log.Info("Loading patch " + ns.GetName() + ", " + patchName)
+		if err := i.client.Get(context.Background(), kube_types.NamespacedName{Namespace: ns.GetName(), Name: patchName}, containerPatch); err != nil {
+			log.Error(err, "invalid ContainerPatch", "podName", pod.Name, "namespace", pod.Namespace, "name", patchName)
+			return nil, nil, err
+		}
+		sidecarPatches = append(sidecarPatches, containerPatch.Spec.SidecarPatch...)
+		initPatches = append(initPatches, containerPatch.Spec.InitPatch...)
+	}
+	return sidecarPatches, initPatches, nil
+}
+
+// applyCustomPatches applies the block of patches to the given container and returns a new,
+// patched container. If patch list is empty, the same unaltered container is returned.
+func (i *KumaInjector) applyCustomPatches(container kube_core.Container, patches []mesh_k8s.JsonPatchBlock) (kube_core.Container, error) {
+	if patches == nil {
```

Comment:
```suggestion
	if len(patches) == 0 {
```

It's little less error prone from golang's nil empty array weirdness